### PR TITLE
chore: added prettier config

### LINKED ISSUE #646: Add .prettierrc to project
#### Reason/Context

When working on a custom Go preset, ESLint was complaining about listing rules I was violating. Somehow I couldn't configure VSCode's prettier/formatter integration to stop formatting code when there is no prettierrc.

Overall I think it would help contributors quite a bit to have prettier on the project. Having to manually deal with formatting is a pain and prettier is pretty standard and well supported.

#### Description

- Add prettier as a dependency
- Add a `.prettierrc`
- Update `.eslintrc` to disable rules that are handled by prettier.

### PR DESCRIPTION
**Pull request by akshatnema**
**Description**

Added `.prettierrrc` file to add the Prettier configuration along with ESlint configuration for typescript.

**Related issue(s)**
Resolves #646

### CODE REVIEW

**Comment by akshatnema** [comment]
@jonaslagoni I added the script `npm run format` in `package.json` file to format the TS files and linked it with ESlint. I made a config for prettier but want a review from you in order to add extra options to the file. Also, I haven't formatted the files in the repo using the script. Let's finalise the config first and then we will update all the files according to the configuration.

Prettier Config Options - https://prettier.io/docs/en/options.html

**Review by jonaslagoni — COMMENTED** [review]
This looks like a good start to me, we can always add more formatting rules for later 👍 

I think it makes sense to at least wait 2 weeks until we merge this as we are still in the holiday season and no rush, so everyone has a chance to review it 🙂

**Comment by akshatnema** [comment]
@jonaslagoni should I format the file according to the prettier and update the PR OR no changes are required right now to format files?

**Comment by jonaslagoni** [comment]
> @jonaslagoni should I format the files according to the prettier and update the PR OR no changes are required right now to format files?

Let's keep it like this until it's time to merge, otherwise you will get a tons of merge conflicts to fix, which does not make much sense 🙂

**Comment by jonaslagoni** [comment]
cc @magicmatatjahu @Samridhi-98 @leigh-johnson @ron-debajyoti in case you have any preferences, as this is going to be merged soon.

**Comment by magicmatatjahu** [comment]
It can be merged whenever you want 😆

**Comment by akshatnema** [comment]
@jonaslagoni are you going to run the script `npm run format` before merge or after merge?

**Comment by jonaslagoni** [comment]
> @jonaslagoni are you going to run the script `npm run format` before merge or after merge?

I am not sure yet, gonna ping you if i need you to run the command once before release :)

**Comment by jonaslagoni** [comment]
I am pretty sure we will apply this as the last change before merging `next` into `master` as part of version 1. That way we are sure there won't be or at least to a very minimum, affect any existing PRs or changes.

File: package.json
Comment (jonaslagoni) [inline package.json:90]:
```suggestion
    "prepublishOnly": "npm run build:prod && npm run generate:readme:toc && npm run format",
```

**Review by jonaslagoni — APPROVED** [review]
/rtm

**Pull request merged**
